### ISSUE TITLE
Error testing lock: daemonsets.extensions \"kube-system\" not found"

### ISSUE THREAD

**Issue by garethmorris78** [weaveworks/kured#34]
Running K8s 1.11.2 on Azure Kubernetes Service and getting the errors with RBAC applied ( across all pods in the daemonset); using the yaml manifests under the master branch of kured repo. 

```
time="2018-10-19T13:13:20Z" level=info msg="Kubernetes Reboot Daemon: master-b86c60f"
time="2018-10-19T13:13:20Z" level=info msg="Node ID: aks-node"
time="2018-10-19T13:13:20Z" level=info msg="Lock Annotation: kured/kube-system:weave.works/kured-node-lock"
time="2018-10-19T13:13:20Z" level=info msg="Reboot Sentinel: /var/run/reboot-required every 24h0m0s"
time="2018-10-19T13:13:20Z" level=fatal msg="Error testing lock: daemonsets.extensions \"kube-system\" not found"
```
```
kubectl get clusterrole kured
NAME    AGE
kured   37m
```
```
kubectl get clusterrolebinding kured
NAME    AGE
kured   38m
```

```
kubectl get role kured -n kube-system
NAME    AGE
kured   41m
```
```
 kubectl get rolebinding kured -n kube-system
NAME    AGE
kured   42m
```

kured is now referenced via https://docs.microsoft.com/en-us/azure/aks/concepts-security#node-security so would be good to get it working on aks :)

**Comment by bboreham** [comment]
From your logs, and [this line in the code](https://github.com/weaveworks/kured/blob/master/cmd/kured/main.go#L285), it is trying to access a DaemonSet called `kube-system` in the `kured` namespace.

Which suggests you have some parameters the wrong way round?

**Comment by garethmorris78** [comment]
error was due to me passing in the defaults within the args instead of leaving them commented out. 

i.e. 
Yaml causing the error:
```
apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: kured            
  namespace: kube-system
spec:
  selector:
    matchLabels:
      name: kured
  updateStrategy:
   type: RollingUpdate
  template:
    metadata:
      labels:
        name: kured
    spec:
      serviceAccountName: kured
      containers:
        - name: kured
          image: quay.io/weaveworks/kured:1.0.0
          imagePullPolicy: IfNotPresent
          command:
            - /usr/bin/kured
          args:
#            - --alert-filter-regexp=^RebootRequired$
            - --ds-name=kured
            - --ds-namespace=kube-system
            - --lock-annotation=weave.works/kured-node-lock
            - --period=24h
#            - --prometheus-url=http://prometheus.monitoring.svc.cluster.local
            - --reboot-sentinel=/var/run/reboot-required
            - --slack-hook-url=https://hooks.slack.com/[redacted]
            - --slack-username=dev
# NO USER SERVICEABLE PARTS BEYOND THIS POINT
          env:
            # Pass in the name of the node on which this pod is scheduled
            # for use with drain/uncordon operations and lock acquisition
            - name: KURED_NODE_ID
              valueFrom:
                fieldRef:
                  fieldPath: spec.nodeName
          volumeMounts:
            # Needed for two purposes:
            # * Testing for the existence of /var/run/reboot-required
            # * Accessing /var/run/dbus/system_bus_socket to effect reboot
            - name: hostrun
              mountPath: /var/run
      restartPolicy: Always
      volumes:
        - name: hostrun
          hostPath:
            path: /var/run
```
comment out defaults and now working yaml:

```
apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: kured            
  namespace: kube-system
spec:
  selector:
    matchLabels:
      name: kured
  updateStrategy:
   type: RollingUpdate
  template:
    metadata:
      labels:
        name: kured
    spec:
      serviceAccountName: kured
      containers:
        - name: kured
          image: quay.io/weaveworks/kured:1.0.0
          imagePullPolicy: IfNotPresent
          command:
            - /usr/bin/kured
          args:
#           - --alert-filter-regexp=^RebootRequired$
#           - --ds-name=kured
#           - --ds-namespace=kube-system
#           - --lock-annotation=weave.works/kured-node-lock
            - --period=24h
#           - --prometheus-url=http://prometheus.monitoring.svc.cluster.local
#           - --reboot-sentinel=/var/run/reboot-required
            - --slack-hook-url=https://hooks.slack.com/[redacted]
            - --slack-username=dev
# NO USER SERVICEABLE PARTS BEYOND THIS POINT
          env:
            # Pass in the name of the node on which this pod is scheduled
            # for use with drain/uncordon operations and lock acquisition
            - name: KURED_NODE_ID
              valueFrom:
                fieldRef:
                  fieldPath: spec.nodeName
          volumeMounts:
            # Needed for two purposes:
            # * Testing for the existence of /var/run/reboot-required
            # * Accessing /var/run/dbus/system_bus_socket to effect reboot
            - name: hostrun
              mountPath: /var/run
      restartPolicy: Always
      volumes:
        - name: hostrun
          hostPath:
            path: /var/run
```

**Comment by awh** [comment]
Fixed by #5.
